cookbook/case_studies/ml_training/nlp_processing/word2vec_and_lda.py

```diff
@@ -0,0 +1,331 @@
+# - ``update_every``: Number of documents to be iterated through for each update.
+# - ``random_state``: seed for reproducibility
+@dataclass_json
+@dataclass
+class LDAModelHyperparams(object):
+    """
+    These are the lda model hyper parameters which can be set for training.
+    """
+
+    num_topics: int = 5
+    alpha: str = "auto"
+    passes: int = 10
+    chunksize: int = 100
+    update_every: int = 1
+    random_state: int = 100
+
+# %%
+# Training
+# ========
+#
+# This task initialises and trains the Word2Vec model on the preprocessed corpus using
+# the hyperparameters defined in the dataclass. It prints the training loss to the user and
+# serialises the model to disk.
+@task
+def train_word2vec_model(
+    training_data: List[List[str]], hyperparams: Word2VecModelHyperparams
+) -> model_file:
+
+    model = Word2Vec(
+        training_data,
+        min_count=hyperparams.min_count,
+        workers=hyperparams.workers,
+        vector_size=hyperparams.vector_size,
+        compute_loss=hyperparams.compute_loss,
+    )
+    training_loss = model.get_latest_training_loss()
+    print(f"training loss: {training_loss}")
+    file = "word2vec.model"
+    model.save(file)
+    return (file,)
+
+
+# %%
+# Here we train the LDA model using the hyperparameters from the dataclass.
+# Before this we need to transform the documents to a vectorized form and compute the frequency of each word
+# to generate a bag of word corpus for the LDA model to train on. We also need to create a mapping
+# from word IDs to words as input to the LDA model for training. The words in each topic are printed
+# out once the model has finished training. Finally, the model is serialised to disk.
+@task
+def train_lda_model(
+    corpus: List[List[str]], hyperparams: LDAModelHyperparams
+) -> model_file:
+    id2word = Dictionary(corpus)
+    bow_corpus = [id2word.doc2bow(doc) for doc in corpus]
+    id_words = [[(id2word[id], count) for id, count in line] for line in bow_corpus]
+    print(f"sample of bag of words generated: {id_words[:2]}")
+    lda = LdaModel(
+        corpus=bow_corpus,
+        id2word=id2word,
+        num_topics=hyperparams.num_topics,
+        alpha=hyperparams.alpha,
+        passes=hyperparams.passes,
+        chunksize=hyperparams.chunksize,
+        update_every=hyperparams.update_every,
+        random_state=hyperparams.random_state,
+    )
+    pprint(f"Shows words in each topic: {lda.print_topics(num_words=5)}")
+    file = "lda.model"
+    lda.save(file)
+    return (file,)
+
+
+# %%
+# Word Similarities
+# ========
+#
+# The model is deserialised from disk and used to compute the top 10 most similar
+# words in a corpus to a given word (we will use the word `computer` when running
+# the workflow to output the similar words). Note that since the model is trained
+# on a small corpus, some of the relations might not be so clear.
```

Comment:
```suggestion
# Deserialize the model from disk and compute the top 10 similar
# words to the given word in the corpus (we will use the word `computer` when running
# the workflow to output similar words). Note that since the model is trained
# on a small corpus, some of the relations might not be clear.
```